### PR TITLE
Add security tests for create key api

### PR DESCRIPTION
**Pull request by HawieLong**
Add several security tests for create key api

Signed-off-by: Long Hanyu <hanyu.long@intel.com>

### CODE REVIEW

File: ehsm_kms_service/test/security_test/createkey.py [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+
+def test_createkey_with_invalid_appid(test_url, invalid_appid_params, appkey_params, timestamp_id_params, keyspec_params, origin_params):
+    params = init_params(invalid_appid_params, appkey_params, timestamp_id_params, keyspec_params, origin_params)
+    resp = requests.post(url=test_url + "/ehsm?Action=CreateKey", data=json.dumps(params), headers=_utils_.headers)
+    print(resp.text)
+    resp_json = json.loads(resp.text)
+    assert 200 != resp_json["code"]
+
+
+def test_createkey_with_invalid_appkey(test_url, appid_params, invalid_appkey_params, timestamp_id_params, keyspec_params, origin_params):
```
Comment (syan10) [inline ehsm_kms_service/test/security_test/createkey.py:132]: test_createkey_with_invalid_appkey -> test_createkey_with_invalid_apikey

File: ehsm_kms_service/test/security_test/createkey.py [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+def test_wrong_sign_request(test_url, appid_params, appkey_params, timestamp_params, keyspec_params, origin_params):
+    params = init_params_with_invalid_sign(appid_params, appkey_params, timestamp_params, timestamp_params, keyspec_params, origin_params)
+    resp = requests.post(url=test_url + "/ehsm?Action=CreateKey", data=json.dumps(params), headers=_utils_.headers)
+    print(resp.text, timestamp_params)
+    resp_json = json.loads(resp.text)
+    assert 200 != resp_json["code"]
+
+#  Test timestamp cache whether available within 20 mins (NONCE_CACHE_TIME)
+def test_uaing_createkey_api_for_20mins(test_url, appid_params, appkey_params, origin_params):
+    timestamp = str(int((time.time()+599) * 1000))
```
Comment (syan10) [inline ehsm_kms_service/test/security_test/createkey.py:180]: typo? uaing

File: ehsm_kms_service/test/security_test/createkey.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+import hmac
+from hashlib import sha256
+from collections import OrderedDict
+import urllib.parse
+import pytest
+from conftest import test_url
+
+import _utils_
+
+common_invalid_params = ["123", "000", "aaaaaa", "**.,][][", "!=1 or", "12.13"]
```
Comment (yang8621) [inline ehsm_kms_service/test/security_test/createkey.py:16]: not used?

File: ehsm_kms_service/test/security_test/createkey.py (hunk 1 above)
Comment (HawieLong) [reply ehsm_kms_service/test/security_test/createkey.py]: fixed

File: ehsm_kms_service/test/security_test/createkey.py (hunk 2 above)
Comment (HawieLong) [reply ehsm_kms_service/test/security_test/createkey.py:178]: fixed

**Comment by yang8621** [comment]
image

**Review by yang8621 — APPROVED** [review]

File: ehsm_kms_service/test/security_test/createkey.py [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+
+#  Max timestamp diff is 10 mins set in eHSM
+#  Thus the nonce is set to twice timestamp diff
+#  MAX_TIME_STAMP_DIFF = 10 * 60 * 1000
+#  NONCE_CACHE_TIME = MAX_TIME_STAMP_DIFF * 2
+NONCE_CACHE_TIME_MINS = 20
+
+
+def init_params(appid, appkey, timestamp_id, keyspec, origin, sign=None):
+    payload = OrderedDict()
```
Comment (syan10) [inline ehsm_kms_service/test/security_test/createkey.py:24]: please replace the "appkey" to "apikey" in the whole file.

File: ehsm_kms_service/test/security_test/_utils_.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    return params
+
+def check_result(res, action):
+    res_json = json.loads(res.text)
+    if(res_json['code'] == 200):
+        print("%s successfully \n" %(action))
+        return True
+    else:
+        print("%s failed, error message: %s \n" %(action, res_json["message"]))
+        return False
```
Comment (syan10) [inline ehsm_kms_service/test/security_test/_utils_.py:38]: seems this _utils_ file only used for import the headers "_utils_.headers", if so, please delete the file, and put the headers variable in the test file.

File: ehsm_kms_service/test/security_test/createkey.py (hunk 2 above)
Comment (syan10) [reply ehsm_kms_service/test/security_test/createkey.py:178]: test_using_createkey_api_for_20mins -> test_replay_attack_for_createKey_api

Comment (HawieLong) [reply ehsm_kms_service/test/security_test/createkey.py:178]: fixed

File: ehsm_kms_service/test/security_test/createkey.py (hunk 4 above)
Comment (HawieLong) [reply ehsm_kms_service/test/security_test/createkey.py:24]: fixed

**Review by syan10 — APPROVED** [review]

**Pull request merged**